packages/offix-client/src/config/ApolloOfflineClientConfigValidator.ts

```diff
@@ -0,0 +1,17 @@
+import { ApolloOfflineClientConfig } from "./ApolloOfflineClientConfig";
+import { ConfigError } from "./ConfigError";
+
+export default function(config: ApolloOfflineClientConfig) {
+    checkFields(config, [
+        "conflictStrategy", "conflictProvider", "cacheStorage",
```

Comment:
`conflictStrategy` and `conflictProvider` have fallbacks set when the client Conflig is created, 
so I think those two can be removed. 

`cacheStorage` and `offlineStorage` are also created before this, if nothing was defined, but I think it is still worthwhile keeping those in there. The nice thing about how you have done things is that it is dynamic, so we can add or remove things easily if we need to.

Reply:
Thank you. That's what I was going for. I tried to generate the list of required fields at runtime from the given type, i.e `ApolloOfflineClientConfig`, to no avail. So, I settled for this instead.